src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1740,6 +1742,32 @@ static bool processOnChangeConditionForUpdateContext
 
 /* ****************************************************************************
 *
+* setMetadataForFlags -
+*/
+static void setMetadataForFlags(ContextElementResponse* notifyCerP)
+{
+  for (unsigned int ix = 0; ix < notifyCerP->contextElement.contextAttributeVector.size(); ix++)
+  {
+    ContextAttribute* caP = notifyCerP->contextElement.contextAttributeVector[ix];
+
+    if (caP->onUpdate)
+    {
+      Metadata* newMdP = new Metadata(NGSI_MD_NOTIF_ONUPDATE, DEFAULT_ATTR_BOOL_TYPE, true);
+      caP->metadataVector.push_back(newMdP);
+    }
+
+    if (caP->onChange)
+    {
+      Metadata* newMdP = new Metadata(NGSI_MD_NOTIF_ONCHANGE, DEFAULT_ATTR_BOOL_TYPE, true);
+      caP->metadataVector.push_back(newMdP);
+    }
+  }
```

Comment:
I believe the names (onUpdate and onChange) confuses me a little here.
Can both onChange and onUpdate be true at the same time?
I ask this as adding an `else` may make the code easier to read and avoid a possible bug.